iothub/device/src/Transport/AmqpIot/AmqpIotExceptionAdapter.cs

```diff
@@ -19,6 +19,13 @@ internal static Exception ConvertToIotHubException(Exception exception)
             {
                 return new UnauthorizedException(exception.Message, exception);
             }
+            else if (exception is OperationCanceledException)
```

Comment:
```suggestion
else if (exception is OperationCanceledException
    && exception.InnerException is AmqpException innerAmqpException
    && innerAmqpException != null)
```

Reply:
Will this change any behavior in any way?

Reply:
I worked backwards somehow :D Thanks for the explanation in the details of the PR. That explains it all!